### PR TITLE
Add ContainerRef in kustomization type

### PR DESCRIPTION
**Pull request by Liujingfang1**

### CODE REVIEW

File: pkg/types/kustomization.go [hunk 1]
```diff
@@ -68,6 +68,12 @@ type Kustomization struct {
 
 	// Variables which will be substituted at runtime
 	Vars []Var `json:"vars,omitempty" yaml:"vars,omitempty"`
+
+	// If set to true, all images need to have tags
+	RequireTag bool `json:"requireTag,omitempty" yaml:"requireTag,omitempty"`
+
+	// Container Reference lists for changing image tags
+	CRs []ContainerRef `json:"cr,omitempty" yaml:"cr,omitempty"`
```
Comment (mengqiy) [inline pkg/types/kustomization.go]: `CRs` should be consistent with json|yaml tag `cr`.
Both have `s` or neither has `s`.

File: pkg/types/kustomization.go [hunk 2]
```diff
@@ -136,3 +142,16 @@ type DataSources struct {
 	// i.e. a Docker .env file or a .ini file.
 	EnvSource string `json:"env,omitempty" yaml:"env,omitempty"`
 }
+
+// ImageTag contains an image and a new tag, which will replace the original tag
```
Comment (monopole) [inline pkg/types/kustomization.go]: // ImageTag describes data for manipulating image tags.

File: pkg/types/kustomization.go [hunk 3]
```diff
@@ -136,3 +142,16 @@ type DataSources struct {
...
+type ImageTag struct {
+	// Name is the name for an image without tag
```
Comment (monopole) [inline pkg/types/kustomization.go]: // Name is a tag-less image name.

File: pkg/types/kustomization.go [hunk 4]
```diff
@@ -136,3 +142,16 @@ type DataSources struct {
...
+	Name string `json:"name,omitempty" yaml:"name,omitempty"`
+
+	// Tag will replace the original tag associated with Name at runtime
+	Tag string `json:"tag,omitempty" yaml:"tag,omitempty"`
```
Comment (monopole) [inline pkg/types/kustomization.go]: suggest calling this NewTag

// NewTag is the value to use in replacing the original image tag.

File: pkg/types/kustomization.go [hunk 5]
```diff
@@ -136,3 +142,16 @@ type DataSources struct {
...
+
+	// Env, if specified, the value in the environment variable will replace the original
+	// tag associated with Name at runtime
+	Env string `json:"env,omitempty" yaml:"env,omitempty"`
```
Comment (monopole) [inline pkg/types/kustomization.go]: i don't think we should support this.  It breaks accountability, in that one's configuration isn't entirely stored in a file.  If we allow env var override here, it suggests we should do so for nameprefix, commonlabel etc.

suggest instead we provide an "edit" command to easily change the NewTag field in a kustomization file.

**Pull request merged**